source/common/http/conn_manager_impl.cc

```diff
@@ -790,6 +811,10 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
                                connection_manager_.read_callbacks_->connection().dispatcher());
   request_headers_ = std::move(headers);
   filter_manager_.requestHeadersInitialized();
+  if (request_header_timer_ != nullptr) {
+    request_header_timer_->disableTimer();
+    request_header_timer_ = nullptr;
+  }
```

Comment:
Should we consider instead: request_header_timer_.reset();

Behavior is equivalent.  There is no need to call disable before deleting the timer.

Reply:
Fixed.

Reply:
I did this and it makes testing more difficult since detecting that the timer is deleted is non-trivial.

Reply:
I think your current tests only manage to cover the case of disable while waiting for min, not while scaling max.  This is related to your tests depending on MockTimer::disable instead of methods of the RangeTimer interface.  I'm not sure what advice to provide about how to best test this.  Having some way to query is this timer exists in the connection may help.  As for detecting mock timer deletion, you could add a mock call to a "Destroy" mock method from the mock object's destructor.